GDAXSharp/WebSocket/WebSocket.cs

```diff
@@ -0,0 +1,135 @@
+﻿using GDAXSharp.Network;
+using GDAXSharp.Network.Authentication;
+using GDAXSharp.Shared.Types;
+using GDAXSharp.Shared.Utilities.Clock;
+using GDAXSharp.Shared.Utilities.Extensions;
+using GDAXSharp.WebSocket.Models.Request;
+using GDAXSharp.WebSocket.Models.Response;
+using Newtonsoft.Json;
+using SuperSocket.ClientEngine;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+using System.Threading;
+using WebSocket4Net;
+
+namespace GDAXSharp.WebSocket
+{
+    public class WebSocket : AbstractRequest, IWebSocket
+    {
+        private const string ApiUri = "wss://ws-feed.gdax.com";
+
+        private const string SandBoxApiUri = "wss://ws-feed-public.sandbox.gdax.com";
+
+        private CancellationToken CancellationToken { get; set; }
+        private ProductType[] ProductTypes { get; set; }
+        private WebSocket4Net.WebSocket WebSocketFeed { get; set; }
+
+        public WebSocket(IAuthenticator authenticator, IClock clock, CancellationToken cancellationToken, bool sandBox = false) : base(authenticator, clock, sandBox)
+        {
+            CancellationToken = cancellationToken;
+        }
+
+        public void GetTickerChannel(params ProductType[] productTypes)
+        {
+            ProductTypes = productTypes;
+            if (ProductTypes.Length == 0)
+            {
+                throw new ArgumentException("You must specify at least one product type");
+            }
+
+            var socketUrl = SandBox ? SandBoxApiUri : ApiUri;
+
+            WebSocketFeed = new WebSocket4Net.WebSocket(socketUrl);
+            WebSocketFeed.Closed += WebSocket_Closed;
+            WebSocketFeed.Error += WebSocket_Error;
+            WebSocketFeed.MessageReceived += WebSocket_MessageReceived;
+            WebSocketFeed.Opened += WebSocket_Opened;
+            WebSocketFeed.Open();
+        }
+
+        private void WebSocket_Opened(object sender, EventArgs e)
+        {
+            var timeStamp = Clock.GetTime().ToTimeStamp();
+            var json = JsonConvert.SerializeObject(new TickerChannel
```

Comment:
@dougdellolio @chrisw000 would you be willing to look at this? Currently it's not changeable by the users. So I was thinking, perhaps a part of could be a parameter. Users would be able to specify which channels they want to subscribe to.

Reply:
Don't see why not. Have a flag enum to represent each channel... . allow use to pass an channels emum value into GetTickerChannel (default to All) - then use that value to determine which Channel objects to add into the List<Channel> when setting up the WebSocketFeed.

If the Channel objects property Namewere also this enum type (with correct serialization info on the enum) it would just be a loop to add each channel being passed in?